File: easy/Set-Intersection/Set Intersection.cs
using System.IO;
using System.Collections.Generic;

class Program
{
    static void Main(string[] args)
    {
        using (StreamReader reader = File.OpenText(args[0]))
        while (!reader.EndOfStream)
        {
            string line = reader.ReadLine();
            if (null == line)
                continue;
            // do something with line
            ShowIntersection(line); 
        }
    }
    
    private static void ShowIntersection(string line) {
        int pos = line.IndexOf(";");
        string firstStr = line.Substring(0, pos);
        string[] second = line.Substring(pos + 1).Split(',');
        string result = "";
        for(int i=0;i<second.Length;i++) if(firstStr.Contains(second[i])) result = result + "," + second[i];
        if (result.Length>0)  System.Console.WriteLine(result.Substring(1));
        else System.Console.WriteLine();
    }
}

File: easy/Set-Intersection/Set Intersection.js
import java.io.*;
public class Main {
    public static void main (String[] args) throws IOException {
        File file = new File(args[0]);
        BufferedReader buffer = new BufferedReader(new FileReader(file));
        String line;
        while ((line = buffer.readLine()) != null) {
            line = line.trim();
            // Process line of input Here
            ShowIntersection(line);
        }
    }
    
    private static void ShowIntersection(String line) {
        int pos = line.indexOf(";");
        String firstStr = line.substring(0, pos);
        String[] second = line.substring(pos + 1).split(",");
        String result = "";
        for(int i=0;i<second.length;i++) if(firstStr.contains(second[i])) result = result + "," + second[i];
        if (result.length()>0)  System.out.println(result.substring(1));
        else System.out.println();
    }
}
